python/gtsam/examples/CombinedImuFactorExample.py

```diff
@@ -0,0 +1,251 @@
+        """
+        Plot poses in values.
+
+        Args:
+            values: The values object with the poses to plot.
+            title: The title of the plot.
+            fignum: The matplotlib figure number.
+                POSES_FIG is a value from the PreintegrationExample
+                which we simply increment to generate a new figure.
+            show: Flag indicating whether to display the figure.
+        """
+        i = 0
+        while values.exists(X(i)):
+            pose_i = values.atPose3(X(i))
+            plot_pose3(fignum, pose_i, 1)
+            i += 1
+        plt.title(title)
+
+        gtsam.utils.plot.set_axes_equal(fignum)
+
+        i = 0
+        while values.exists(B(i)):
+            print("Bias Value {0}".format(i), values.atConstantBias(B(i)))
+            i += 1
+
+        plt.ioff()
+
+        if show:
+            plt.show()
+
+    def run(self,
+            T: int = 12,
+            compute_covariances: bool = False,
+            verbose: bool = True):
+        """
+        Main runner.
+
+        Args:
+            T: Total trajectory time.
+            compute_covariances: Flag indicating whether to compute marginal covariances.
+            verbose: Flag indicating if printing should be verbose.
+        """
+        graph = gtsam.NonlinearFactorGraph()
+
+        # initialize data structure for pre-integrated IMU measurements
+        pim = gtsam.PreintegratedCombinedMeasurements(self.params,
+                                                      self.actualBias)
+
+        num_poses = T  # assumes 1 factor per second
+        initial = gtsam.Values()
+
+        # simulate the loop
+        i = 0  # state index
+        initial_state_i = self.scenario.navState(0)
+        initial.insert(X(i), initial_state_i.pose())
+        initial.insert(V(i), initial_state_i.velocity())
+        initial.insert(B(i), self.actualBias)
+
+        # add prior on beginning
+        self.addPrior(0, graph)
+
+        for k, t in enumerate(np.arange(0, T, self.dt)):
+            # get measurements and add them to PIM
+            measuredOmega = self.runner.measuredAngularVelocity(t)
+            measuredAcc = self.runner.measuredSpecificForce(t)
+            pim.integrateMeasurement(measuredAcc, measuredOmega, self.dt)
+
+            # Plot IMU many times
+            if k % 10 == 0:
+                self.plotImu(t, measuredOmega, measuredAcc)
+
+            if (k + 1) % int(1 / self.dt) == 0:
+                # Plot every second
+                self.plotGroundTruthPose(t, scale=1)
+                plt.title("Ground Truth Trajectory")
+
+                # create IMU factor every second
+                factor = gtsam.CombinedImuFactor(X(i), V(i), X(i + 1),
+                                                 V(i + 1), B(i), B(i + 1), pim)
+                graph.push_back(factor)
+
+                if verbose:
+                    print(factor)
+                    print(pim.predict(initial_state_i, self.actualBias))
```

Comment:
should  there be a `initial_state_i = self.scenario.navState(i)` somewhere? currently `initial_state_i` is only for `i=0`